src/vs/workbench/contrib/replNotebook/browser/replEditorInput.ts

```diff
@@ -51,6 +52,20 @@ export class ReplEditorInput extends NotebookEditorInput {
 	) {
 		super(resource, undefined, 'jupyter-notebook', {}, _notebookService, _notebookModelResolverService, _fileDialogService, labelService, fileService, filesConfigurationService, extensionService, editorService, textResourceConfigurationService, customEditorLabelService);
 		this.isScratchpad = resource.scheme === 'untitled' && configurationService.getValue<boolean>(NotebookSetting.InteractiveWindowPromptToSave) !== true;
+		this.label = this.createEditorLabel(resource);
+	}
+
+	private createEditorLabel(resource: URI | undefined): string {
+		if (!resource) {
+			return 'REPL';
+		}
+
+		const match = new RegExp('Untitled-(\\d+)\.').exec(resource.path);
```

Comment:
If you want this to avoid triggering on saved files that have that name you could check the scheme

Reply:
good call